File: python/examples/calls/override.py
from mirascope import llm


@llm.call("openai/gpt-5-mini")
def recommend_book(genre: str):
    return f"Please recommend a book in {genre}."


# Override the model at runtime
with llm.model("anthropic/claude-sonnet-4-5", temperature=0.9):
    response = recommend_book("fantasy")
    print(response.pretty())

File: typescript/examples/calls/override.ts
/**
 * Runtime model override example.
 *
 * Demonstrates overriding the model at runtime using withModel.
 *
 * Run with: bun run example examples/calls/override.ts
 */
import { llm } from "mirascope";

const recommendBook = llm.defineCall<{ genre: string }>({
  model: "openai/gpt-4o-mini",
  template: ({ genre }) => `Please recommend a book in ${genre}.`,
});

// Override the model at runtime
const response = await llm.withModel(
  "anthropic/claude-haiku-4-5",
  { temperature: 0.9 },
  async () => {
    return recommendBook({ genre: "fantasy" });
  },
);

console.log(response.text());
